modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -63,17 +63,48 @@ void blobShapeFromTensor(const tensorflow::TensorProto &tensor, MatShape& shape)
     {
         const tensorflow::TensorShapeProto &_shape = tensor.tensor_shape();
         int i, n = _shape.dim_size();
-        shape.resize(n);
+        if (n)
+        {
+            shape.resize(n);
 
-        for (i = 0; i < n; i++)
-            shape[i] = (int)_shape.dim(i).size();
+            for (i = 0; i < n; i++)
+                shape[i] = (int)_shape.dim(i).size();
+        }
+        else
+            shape.resize(1, 1);  // Scalar.
     }
     else
     {
         CV_Error(Error::StsError, "Unknown shape of input tensor");
     }
 }
 
+static Mat getTensorContent(const tensorflow::TensorProto &tensor)
+{
+    std::string content = tensor.tensor_content();
+    switch (tensor.dtype())
+    {
+        case tensorflow::DT_FLOAT:
+            return Mat(1, content.size() / sizeof(float), CV_32FC1, (void*)content.c_str()).clone();
+        case tensorflow::DT_DOUBLE:
+            return Mat(1, content.size() / sizeof(double), CV_64FC1, (void*)content.c_str()).clone();
+        case tensorflow::DT_HALF:
+        {
+            const RepeatedField<int32_t>& field = tensor.half_val();
+            Mat halfs(1, field.size(), CV_32SC1, (void*)field.data());
+            halfs.convertTo(halfs, CV_16UC1);
+
+            Mat values(halfs.size(), CV_16SC1, halfs.data);
+            convertFp16(values, values);
```

Comment:
Be careful with pseudo-inplace processing. A lot of functions don't support this well, also there is almost no tests for such functionality.

Reply:
Thanks! Fixed.